### PR TITLE
Etcd

### PR DESCRIPTION
**Pull request by asuresh4**

### CODE REVIEW

**Comment by molner** [comment]
Python plugin is always installed when using the SFx collectd. You can remove the blurb at the .conf file about it "sometimes being installed with collectd". Should instead mention that it's always installed with SFx installer but if you're using non-SFx you might need to install it. (But they're already having to do so for our metadata plugin if it wasn't already there for some reason)

File: collectd-etcd/README.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+|-----------|----------------|
+| collectd  |  4.9 or later  |
+| python | 2.7 or later |
+| etcd | 2.0.8 or later |
+| Python plugin for collectd | (included with [SignalFx collectd agent](https://github.com/signalfx/integrations/tree/master/collectd)[](sfx_link:sfxcollectd)) |
+
+
+### INSTALLATION
+
+1. Download the [collectd-etcd Python module](https://github.com/signalfx/collectd-etcd).
```
Comment (charliesignalfx) [inline collectd-etcd/README.md]: You might want to tell the customer where to download or at least where to copy this directory.  Your example config correctly tells them to put it in `/usr/share/collectd...`

File: collectd-etcd/README.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+1. Restart collectd.
+
+### CONFIGURATION
+
+Using the example configuration file [10-etcd.conf](https://github.com/signalfx/integrations/tree/master/collectd-etcd/10-etcd.conf) as a guide, provide values for the configuration options listed below that make sense for your environment and allow you to connect to the etcd members
+
+| configuration option | definition | example value |
+| ---------------------|------------|---------------|
+| ModulePath | Path on disk where collectd can find this module. | "/opt/collectd-etcd" |
```
Comment (charliesignalfx) [inline collectd-etcd/README.md]: I would change the example value to match the example config

File: collectd-etcd/10-etcd.conf [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+#     - ssl_ca_certs - path to file
+#     - EnhancedMetrics - whether or not to include optional metrics
+#     - IncludeMetric - add an individual optional metric
+#     - ExcludeMetric - exclude and individual optional metric
+#   Optionally, change any of the collectd settings according to your
+#   requirements.
+
+LoadPlugin python
+<Plugin python>
+  ModulePath "/opt/collectd-etcd"
```
Comment (charliesignalfx) [inline collectd-etcd/10-etcd.conf]: I would change the module path to /usr/share/collectd/collectd-etcd

File: collectd-etcd/10-etcd.conf [hunk 4]
```diff
@@ -0,0 +1,60 @@
+# Install:
+#   This requires the python plugin. It is sometimes installed with collectd
+#   but not always.  You can install it with:
+#
+#     apt-get install collectd-python
+#
+#   Install the collectd plugin from github:
+#
+#     git clone https://github.com/signalfx/collectd-etcd.git /opt/collectd-etcd
```
Comment (charliesignalfx) [inline collectd-etcd/10-etcd.conf]: change this to /usr/share/collectd/collectd-etcd

File: collectd-etcd/docs/guage.etcd.store.watchers.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+---
+title: Number of Watchers
+brief: Number of watchers
+metric_type: guage
+---
+### Number of Failed Gets
```
Comment (mjtieman) [inline collectd-etcd/docs/guage.etcd.store.watchers.md]: Wrong title

File: collectd-etcd/README.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+- [Configuration](#configuration)
+- [Usage](#usage)
+- [Metrics](#metrics)
+- [License](#license)
+
+### DESCRIPTION
+
+This is the SignalFx etcd plugin. Follow these instructions to install the etcd plugin for collectd.
+
+The [`etcd-collectd`](https://github.com/signalfx/collectd-etcd) plugin collects metrics from etcd members from the three [statistics](https://coreos.com/etcd/docs/latest/v2/api.html#statistics) endpoints and the [metric](https://coreos.com/etcd/docs/latest/v2/metrics.html).
```
Comment (charliesignalfx) [inline collectd-etcd/README.md]: *nit pick* might say "...from the three statistic endpoints and the metric endpoint" or "... from the three statistics and one metric endpoint".  Something just seems sparse about about the way it is phrased now.

File: collectd-etcd/docs/guage.etcd.self.sendpkg.rate.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+---
+title: Package Sent Rate
+brief: Rate at which a leader sends packages
+metric_type: guage
+---
+### Package Sent Rate
+The rate at which a leader receives packages. This metric is reported with the dimension `state` to indicate the current state and `cluster` name.
```
Comment (mjtieman) [inline collectd-etcd/docs/guage.etcd.self.sendpkg.rate.md]: Should be rate at which the leader sends packages

File: collectd-etcd/docs/counter.etcd.store.expire.count.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
+---
+title: Number of Keys / Directories Expired
+brief: Total number of items expired due to TTL
+metric_type: counter
+---
+### Number of Failed Gets
```
Comment (mjtieman) [inline collectd-etcd/docs/counter.etcd.store.expire.count.md]: Wrong title

File: collectd-etcd/docs/counter.etcd.store.compareanddelete.success.md [hunk 9]
```diff
@@ -0,0 +1,7 @@
+---
+title: Number of Successful Compare-and-Deletes
+brief: Total number of successful compare-and-delete operations
+metric_type: counter
+---
+### Number of Failed Compare-and-Deletes
+The total number of successful compare-and-delete operations in the store. This metric is common to all the members of the cluster and therefore, only reported by the leader. This metric is reported with the dimension `state` to indicate the current state and `cluster` name.
```
Comment (mjtieman) [inline collectd-etcd/docs/counter.etcd.store.compareanddelete.success.md]: Should be total number of `failed` ompare-and-delete operations

File: collectd-etcd/docs/counter.etcd.store.compareanddelete.success.md [hunk 10]
```diff
@@ -0,0 +1,7 @@
+---
+title: Number of Successful Compare-and-Deletes
+brief: Total number of successful compare-and-delete operations
+metric_type: counter
+---
+### Number of Failed Compare-and-Deletes
```
Comment (mjtieman) [inline collectd-etcd/docs/counter.etcd.store.compareanddelete.success.md]: Wrong title

File: collectd-etcd/docs/counter.etcd.self.sendappendreq.cnt.md [hunk 11]
```diff
@@ -0,0 +1,8 @@
+---
+title: Append Requests Send
+brief: Total number of append requests sent by a member
+metric_type: counter
+---
+### Append Requests Received
```
Comment (mjtieman) [inline collectd-etcd/docs/counter.etcd.self.sendappendreq.cnt.md]: Wrong title

File: collectd-etcd/README.md [hunk 12]
```diff
@@ -0,0 +1,128 @@
+#![](https://github.com/signalfx/integrations/blob/master/collectd-etcd/img/integrations_etcd.png) etcd
```
Comment (charliesignalfx) [inline collectd-etcd/README.md]: We need a space here between the `#` and the `!`.  I suspect this is copied from Elasticsearch which is also incorrect.  With out the space the text will not be set as a header.  See docker plugin's [readme](https://raw.githubusercontent.com/signalfx/integrations/master/collectd-docker/README.md) as an example

Comment (charliesignalfx) [inline collectd-etcd/README.md]: @asuresh4  Could you please add this space in the elasticsearch readme as well!

File: collectd-etcd/README.md [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+Using the example configuration file [10-etcd.conf](https://github.com/signalfx/integrations/tree/master/collectd-etcd/10-etcd.conf) as a guide, provide values for the configuration options listed below that make sense for your environment and allow you to connect to the etcd members
+
+| configuration option | definition | example value |
+| ---------------------|------------|---------------|
+| ModulePath | Path on disk where collectd can find this module. | "/opt/collectd-etcd" |
+| Host | Define the host name of the etcd member | "localhost" |
+| Port | Define the port at which the member can be reached | "2379" |
+| Cluster | Name of this etcd cluster. | "prod" |
+| EnhancedMetrics | Takes a boolean indicated whether stats from ```/metrics``` are needed | "false" |
+| IncludeMetric | Takes a metric name from the ```/metric``` endpoint that has to included |  |
```
Comment (charliesignalfx) [inline collectd-etcd/README.md]: Please include an example

File: collectd-etcd/README.md [hunk 14]
```diff
@@ -0,0 +1,128 @@
...
+| ExcludeMetric | Takes a metric name from the ```/metric``` endpoint that has to excluded |  |
```
Comment (charliesignalfx) [inline collectd-etcd/README.md]: Please include an example

File: collectd-etcd/README.md [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+    Cluster "test"
+  </Module>
+</Plugin>
+```
+
+### METRICS
+By default, metrics about a member, leader and store are provided. Click [here](./docs) for details. Metrics from ```/metrics``` endpoint can be activated through the configuration file. See [usage](#usage) for details.
+
+#### Metric naming
+`<metric type>.etcd.<endpoint name>.<name of metric>`. This is the format of default metric names reported by the plugin. The optional metrics are named the as available from the `/metrics` endpoint with `_` replaced by `.`.
```
Comment (charliesignalfx) [inline collectd-etcd/README.md]: "The optional metric names are available from the `/metrics` endpoint with `_` replaced by `.`.

File: collectd-etcd/docs/guage.etcd.store.watchers.md [hunk 16]
```diff
@@ -0,0 +1,7 @@
+---
+title: Number of Watchers
+brief: Number of watchers
+metric_type: guage
+---
+### Number of watchers
```
Comment (mjtieman) [inline collectd-etcd/docs/guage.etcd.store.watchers.md]: Capitalize watchers

**Comment by molner** [comment]
Better commit messages please...  https://chris.beams.io/posts/git-commit/

File: collectd-etcd/README.md [hunk 17]
```diff
@@ -0,0 +1,160 @@
...
+LoadPlugin python
+<Plugin python>
+  ModulePath "/usr/share/collectd/collectd-etcd/"
+
+  Import etcd_plugin
+  <Module etcd_plugin>
+    Host "localhost"
+    Port "2379"
+    Interval 10
+    Cluster 1
```
Comment (mjtieman) [inline collectd-etcd/README.md]: Is `Cluster` a string or an int?

Comment (asuresh4) [inline collectd-etcd/README.md]: Yeah this needs to be string I'll change it.

**Comment by mattmolner** [comment]
The last sentence of each description needs to have a period.

**Comment by molner** [comment]
In the 'Configuration' table in integrations/collectd-etcd/README.md we'll need some better example values in there for some of them. There are also a few grammar errors in that chart to address. I will work with you offline to fix these.

**Pull request merged**